proto/storage/report_snapshot.proto

```diff
@@ -25,12 +25,17 @@ message ReportSnapshot {
   ReportType type = 5;
   oneof filter {
     VulnerabilityReportFilters vuln_report_filters = 6;
+    AdHocVulnerabilityReportFilters ondemand_vuln_report_filters = 7;
```

Comment:
If we are going to use on demand this should be `on_demand`.  That said, I think we should pick either `adhoc` or `on demand`.  I don't think we should intermix.